pkg/render/logstorage/linseed/linseed_test.go

```diff
@@ -124,6 +127,47 @@ var _ = Describe("Linseed rendering tests", func() {
 			Expect(cr.Rules).To(ContainElement(secretsRules))
 		})
 
+		It("should support an external elasticsearch endpoint", func() {
+			cfg.ElasticHost = "test-host"
+			cfg.ElasticPort = "443"
+			cfg.ElasticClientSecret = &corev1.Secret{
+				TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      logstorage.ExternalCertsSecret,
+					Namespace: common.OperatorNamespace(),
+				},
+				Data: map[string][]byte{
+					"client.crt": {1, 2, 3},
+					"client.key": {4, 5, 6},
+				},
+			}
+			component := Linseed(cfg)
+			createResources, _ := component.Objects()
+			d, ok := rtest.GetResource(createResources, DeploymentName, render.ElasticsearchNamespace, "apps", "v1", "Deployment").(*appsv1.Deployment)
+			Expect(ok).To(BeTrue(), "Deployment not found")
+
+			// The deployment should have the hash annotation set, as well as a volume and volume mount for the client secret.
+			Expect(d.Spec.Template.Annotations["hash.operator.tigera.io/elastic-client-secret"]).To(Equal("ae1a6776a81bf1fc0ee4aac936a90bd61a07aea7"))
+			Expect(d.Spec.Template.Spec.Volumes).To(ContainElement(corev1.Volume{
+				Name: logstorage.ExternalCertsVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: logstorage.ExternalCertsSecret,
+					},
+				},
+			}))
+			Expect(d.Spec.Template.Spec.Containers[0].VolumeMounts).To(ContainElement(corev1.VolumeMount{
+				Name:      logstorage.ExternalCertsVolumeName,
+				MountPath: "/certs/elasticsearch/mtls",
+				ReadOnly:  true,
+			}))
+
+			// The client secret should also be emitted as a resources, but copied to the destination namespace.
+			s, ok := rtest.GetResource(createResources, logstorage.ExternalCertsSecret, render.ElasticsearchNamespace, "", "v1", "Secret").(*corev1.Secret)
+			Expect(ok).To(BeTrue(), "Secret not copied")
```

Comment:
Should we add some checks for the environment variables that are set when mTLS is enabled ?

Reply:
I think we also need an `ELASTIC_MTLS_ENABLED` check ....